docs/data_classes.rst

```diff
@@ -0,0 +1,427 @@
+
+
+Initializing an NDCube
+----------------------
+To initialize the most basic `~ndcube.NDCube` object, we need is a `numpy.ndarray`-like array containing the data and an APE-14-compliant WCS object (e.g. `astropy.wcs.WCS`) describing the coordinate transformations to and from array-elements.
+Let's create a 3-D array of data with shape ``(4, 4, 5)`` and a WCS object with axes of wavelength, helioprojective longitude, and helioprojective latitude.
+This could represent images of the Sun taken at different wavelengths.
+Remember that due to convention, the order of WCS axes is reversed relative to the data array.
+
+.. code-block:: python
+
+  >>> import astropy.wcs
+  >>> import numpy as np
+
+  >>> from ndcube import NDCube
+
+  >>> # Define data array.
+  >>> data = np.random.rand(4, 4, 5)
+  >>> # Define WCS transformations in an astropy WCS object.
+  >>> wcs = astropy.wcs.WCS(naxis=3)
+  >>> wcs.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> wcs.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> wcs.wcs.cdelt = 0.2, 0.5, 0.4
+  >>> wcs.wcs.crpix = 0, 2, 2
+  >>> wcs.wcs.crval = 10, 0.5, 1
+
+  >>> # Now instantiate the NDCube
+  >>> my_cube = NDCube(data, wcs=wcs)
+
+The data array is stored in ``mycube.data`` while the WCS object is stored in ``my_cube.wcs``.
+The ``.data`` attribute should only be used to access specific raw data values.
+When manipulating/slicing the data it is better to slice the `~ndcube.NDCube` instance as a whole so as to ensure that supporting data --- e.g. coordinates, uncertainties, mask --- remain consistent.
+(See :ref:`cube_slicing`.)
+
+To instantiate a more complex `~ndcube.NDCube` with metadata, a data unit, uncertainties and a mask, we can  the following:
+
+.. code-block:: python
+
+  >>> import astropy.units as u
+  >>> import astropy.wcs
+  >>> import numpy as np
+  >>> from astropy.nddata import StdDevUncertainty
+
+  >>> from ndcube import NDCube
+
+  >>> # Define data array.
+  >>> data = np.random.rand(4, 4, 5)
+
+  >>> # Define WCS transformations in an astropy WCS object.
+  >>> wcs = astropy.wcs.WCS(naxis=3)
+  >>> wcs.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> wcs.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> wcs.wcs.cdelt = 0.2, 0.5, 0.4
+  >>> wcs.wcs.crpix = 0, 2, 2
+  >>> wcs.wcs.crval = 10, 0.5, 1
+
+  >>> # Define mask.  Initially set all elements unmasked.
+  >>> mask = np.zeros_like(data, dtype=bool)
+  >>> mask[0, 0][:] = True  # Now mask some values.
+  >>> # Define uncertainty, metadata and unit.
+  >>> uncertainty = StdDevUncertainty(np.sqrt(np.abs(data)))
+  >>> meta = {"Description": "This is example NDCube metadata."}
+  >>> unit = u.ct
+
+  >>> # Instantiate NDCube with supporting data.
+  >>> my_cube = NDCube(data, wcs=wcs, uncertainty=uncertainty, mask=mask, meta=meta, unit=unit)
+
+Generating `~ndcube.ExtraCoords` and `~ndcube.GlobalCoords` objects and attaching them to your `~ndcube.NDCube` is demonstrated in the :ref:`extra_coords` and :ref:`global_coords` sections.
```

Comment:
This is certainly true for GC, less for EC I guess.

Reply:
This would be true for EC too did NDCube was instantiated with an empty EC object by default.  Unless there's an issue with empty EC coord objects or a performance consideration, we should instantiated NDCube with an empty EC object for consistency and ease-of-use.